Assets/HoloToolkit/Input/Scripts/Utilities/Interactions/TwoHandManipulatable.cs

```diff
@@ -0,0 +1,454 @@
+        /// SetManipulationMode
+        /// </summary>
+        public void SetManipulationMode(TwoHandedManipulation mode)
+        {
+            ManipulationMode = mode;
+        }
+
+        /// <summary>
+        /// Private Methods
+        /// </summary>
+        private void Awake()
+        {
+            m_moveLogic = new TwoHandMoveLogic();
+            m_rotateLogic = new TwoHandRotateLogic(ConstraintOnRotation);
+            m_scaleLogic = new TwoHandScaleLogic();
+        }
+
+        private void Start()
+        {
+            if (HostTransform == null)
+            {
+                HostTransform = transform;
+            }
+        }
+
+        private void Update()
+        {
+            //Update positions of all hands
+            foreach (var key in m_handsPressedInputSourceMap.Keys)
+            {
+                var inputSource = m_handsPressedInputSourceMap[key];
+                Vector3 inputPosition = Vector3.zero;
+                if (inputSource.TryGetGripPosition(key, out inputPosition))
+                {
+                    m_handsPressedLocationsMap[key] = inputPosition;
+                }
+            }
+
+            if (currentState != State.Start)
+            {
+                UpdateStateMachine();
+            }
+        }
+
+        private Vector3 GetInputPosition(InputEventData eventData)
+        {
+            Vector3 result = Vector3.zero;
+            eventData.InputSource.TryGetGripPosition(eventData.SourceId, out result);
+            return result;
+        }
+
+        private void RemoveSourceIdFromHandMap(uint sourceId)
+        {
+            if (m_handsPressedLocationsMap.ContainsKey(sourceId))
+            {
+                m_handsPressedLocationsMap.Remove(sourceId);
+            }
+
+            if (m_handsPressedInputSourceMap.ContainsKey(sourceId))
+            {
+                m_handsPressedInputSourceMap.Remove(sourceId);
+            }
+        }
+
+        /// <summary>
+        /// /// Event Handler receives input from inputSource
+        /// </summary>
+        public void OnInputDown(InputEventData eventData)
+        {
+            // Add to hand map
+            m_handsPressedLocationsMap[eventData.SourceId] = GetInputPosition(eventData);
+            m_handsPressedInputSourceMap[eventData.SourceId] = eventData.InputSource;
+            UpdateStateMachine();
+            eventData.Use();
+        }
+
+        /// <summary>
+        /// Event Handler receives input from inputSource
+        /// </summary>
+        public void OnInputUp(InputEventData eventData)
+        {
+            RemoveSourceIdFromHandMap(eventData.SourceId);
+            UpdateStateMachine();
+            eventData.Use();
+        }
+
+        /// <summary>
+        /// OnSourceDetected Event Handler
+        /// </summary>
+        public void OnSourceDetected(SourceStateEventData eventData)
+        {
+        }
+
+        /// <summary>
+        /// OnSourceLost
+        /// </summary>
```

Comment:
We probably don't need to add summary statements to the interface implementations.  The documentation should get inherited by the doc generator.